admin/sections/config/reranker.tsx

```diff
@@ -0,0 +1,118 @@
+import { toast } from "@/components/ui/use-toast";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useQuery } from "react-query";
+
+export const RerankerConfig = () => {
+  const form = useForm({
+    resolver: zodResolver(CohereRerankerConfigSchema),
+  });
+
+  const { data: config, refetch } = useQuery(
+    "rerankerConfig",
+    getReRankerConfig,
+    {
+      refetchOnWindowFocus: false,
+    },
+  );
+
+  useEffect(() => {
+    if (config) {
+      form.reset(config);
+    }
+  }, [form, config]);
+
+  const submitRerankerForm = async (data: any) => {
+    try {
+      await updateReRankerConfig(data);
+      refetch();
+    } catch (e) {
+      console.error(e);
+      toast({
+        title: "Failed to update reranker configuration",
+        variant: "destructive",
+      });
+    }
+  };
+
+  return (
+    <>
+      <Form {...form}>
+        <form
+          className="space-y-4 mb-4"
+          onBlur={form.handleSubmit(submitRerankerForm)}
+        >
+          <FormField
+            control={form.control}
+            name="use_reranker"
+            render={({ field }) => (
+              <FormItem className="flex flex-row items-center space-x-3 space-y-0">
+                <FormControl>
+                  <Checkbox
+                    checked={field.value}
+                    onCheckedChange={(checked) => {
+                      // Instantly update the form value
+                      field.onChange(checked);
+                      submitRerankerForm(form.getValues());
+                    }}
+                  />
+                </FormControl>
+                <div>
+                  <FormLabel className="font-normal">Use Reranker</FormLabel>
+                  <FormMessage />
+                  <FormDescription>
+                    Use Cohere reranker to rerank the retrieved documents
+                  </FormDescription>
+                  <FormMessage />
+                </div>
+              </FormItem>
+            )}
+          />
+
+          {form.watch("use_reranker") && (
+            <FormField
+              control={form.control}
+              name="cohere_api_key"
+              render={({ field }) => (
+                <FormItem className="ml-6">
+                  <FormLabel>Cohere API Key</FormLabel>
+                  <FormControl>
+                    <PasswordInput {...field} />
+                  </FormControl>
+                  <FormDescription>
+                    You can get your API key from{" "}
+                    <a
+                      href="https://dashboard.cohere.com/api-keys"
+                      target="_blank"
+                      rel="noreferrer"
+                    >
+                      https://dashboard.cohere.com/api-keys
+                    </a>
+                  </FormDescription>
+                  <FormMessage />
+                </FormItem>
```

Comment:
i think if a user is using a reranker, he's more serious, so he should be able to change TOP_K and RERANK_TOP_K

Reply:
For now, i wouldn't generally allow changing TOP_K if the user doesn't select a reranker (complicates setup) - maybe we have an "Advanced" section later for that.

Reply:
That makes sense, i just added config to config RERANK_TOP_K

Reply:
> we just share the TOP_K as the expected number of result - no matter the reranker is enable or not.

After all, i think the point above could confuse the user if we add config for `TOP_K` later. So, i'd just use the TOP_K  and RERANK_TOP_K as the starter idea: If the reranker is enabled, the top_k would be changed to max(top_k, rerank_top_k) (larger value than the rerank_top_k). We should add a note for this if we add config for top_k later.